### PR TITLE
Projectile code cleanup

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Probably TM first since this touches a whole bunch of shit.
Tries to uncurse a bunch of random projectile related shitcode.

- Purged a bunch of unused ammo flags clogging up the list
- AMMO_EXPLOSIVE is now more accurately named AMMO_TARGET_TURF
- Rolled AMMO_ROCKET into AMMO_SNIPER since they literally did the same thing
- Suppressor damage feedback code now actually relates to the suppressed gun flag instead of the entirely unused silenced ammo flag
- Removed the turret projectile interaction with runner evasion regen (now that runners can't attack while evading, theres no reason for this snowflake interaction)
- Moved some damage altering calculations to a more appropriate place, so some slightly more sane calculations and fixes some weird ff interaction bugs/doesn't check things in an inefficient order
- Add a gorillion bits of autodoc
- **Unfucks AMR code**. AMR no longer does funny snowflake signal stuff, meaning it will no longer magically travel through any object in the game, including hull windows. Instead it uses a projectile flag.
## Why It's Good For The Game
Less cursed code.
## Changelog
:cl:
fix: AMR lase mode will no longer completely ignore solid objects
code: Cleaned up a bunch of projectile code
/:cl:

### CODE REVIEW

**Review by ivanmixo — APPROVED** [review]

File: code/modules/codex/entries/ammunition_codex.dm [hunk 1]
```diff
@@ -23,7 +23,7 @@
 		if(ammo.flags_ammo_behavior & AMMO_INCENDIARY)
 			entry.mechanics_text += "Secondary effect: set target on fire.<br>"
 
-		if(ammo.flags_ammo_behavior & AMMO_EXPLOSIVE)
+		if(ammo.flags_ammo_behavior & AMMO_TARGET_TURF)
 			entry.mechanics_text += "Secondary effect: explosion.<br>"
```
Comment (TiviPlus) [inline code/modules/codex/entries/ammunition_codex.dm:27]: In accurate

Comment (TiviPlus) [reply code/modules/codex/entries/ammunition_codex.dm:27]: Below one got fixed though

File: code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm [hunk 2]
```diff
@@ -234,7 +234,7 @@
 		return FALSE
 	if(proj.original_target == xeno_owner && proj.distance_travelled < 2) //Pointblank shot.
 		return FALSE
-	if(!(proj.ammo.flags_ammo_behavior & AMMO_SENTRY) && !xeno_owner.fire_stacks) //We ignore projectiles from automated sources/sentries for the purpose of contributions towards our cooldown refresh; also fire prevents accumulation of evasion stacks
+	if(!xeno_owner.fire_stacks)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm:237]: Should doc this in the body

File: code/modules/projectiles/projectile.dm [hunk 3]
```diff
@@ -43,32 +43,43 @@
...
-
+	///Hit impact sound
 	var/hitsound = null
-	var/datum/ammo/ammo //The ammo data which holds most of the actual info.
-
-	var/def_zone = BODY_ZONE_CHEST	//So we're not getting empty strings.
-
+	///The ammo data which holds most of the actual info
+	var/datum/ammo/ammo
+	///target def zone
```
Comment (TiviPlus) [inline code/modules/projectiles/projectile.dm:52]: What it used for though

File: code/modules/projectiles/projectile.dm [hunk 4]
```diff
@@ -43,32 +43,43 @@
...
-	var/turf/original_target_turf = null // the original target's starting turf
-	var/atom/firer 		 = null // Who shot it
-
-	var/list/atom/movable/uncross_scheduled = list() // List of border movable atoms to check for when exiting a turf.
-
+	///the pixel y location of the tile that the player clicked. Default is the center
+	var/p_y = 16
+	//Pixel location in absolute coordinates. This is (((x - 1) * 32) + 16 + pixel_x)
+	//These values are floats, not integers. They need to be converted through CEILING or such when translated to relative pixel coordinates.
+	///Projectile X coordinate
```
Comment (TiviPlus) [inline code/modules/projectiles/projectile.dm:60]: May want to use 
/**
 *
 */

File: code/modules/projectiles/projectile.dm [hunk 5]
```diff
@@ -43,32 +43,43 @@
...
+	///The object which shot us
+	var/atom/shot_from 	 = null //todo: Shot_from and firer are heavily misused, need a cleanup, for example a mech should have firer = the occupant, shot_from as the mech
+	///the projectile's starting turf
+	var/turf/starting_turf = null
+	///the original target clicked
+	var/atom/original_target = null
+	///the original target's starting turf
+	var/turf/original_target_turf = null
+	///The atom (usually mob) who shot it
+	var/atom/firer = null
```
Comment (TiviPlus) [inline code/modules/projectiles/projectile.dm:74]: Should just strip the = bull here

File: code/modules/projectiles/projectile.dm [hunk 6]
```diff
@@ -81,19 +92,26 @@
...
 	//Fired processing vars
+	///Last movement time
 	var/last_projectile_move = 0
+	///How many movements it needs to make in the next tick
 	var/stored_moves = 0
-	var/dir_angle //0 is north, 90 is east, 180 is south, 270 is west. BYOND angles and all.
-	var/x_offset //Float, not integer.
+	//0 is north, 90 is east, 180 is south, 270 is west. BYOND angles and all
+	var/dir_angle
+	//Float, not integer
```
Comment (TiviPlus) [inline code/modules/projectiles/projectile.dm:109]: Should roll this into the autodoc lines

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Nice batch of overdue changes

File: code/modules/projectiles/projectile.dm (hunk 4 above)
Comment (Lumipharon) [reply code/modules/projectiles/projectile.dm:60]: top 2 lines aren't autodoc, they relate to both vars. I didn't really want to copy paste all the text for both

**Pull request merged**